orgparse/__init__.py

```diff
@@ -130,16 +129,31 @@ def load(path: Union[str, Path, TextIO], env: Optional[OrgEnv]=None) -> OrgNode:
 
     """
     orgfile: TextIO
-    if isinstance(path, (str, Path)):
-        # Use 'with' to close the file inside this function.
-        with codecs.open(str(path), encoding='utf8') as orgfile:
-            lines = (l.rstrip('\n') for l in orgfile.readlines())
-        filename = str(path)
-    else:
-        orgfile = path
-        lines = (l.rstrip('\n') for l in orgfile.readlines())
-        filename = path.name if hasattr(path, 'name') else '<file-like>'
-    return loadi(lines, filename=filename, env=env)
+
+    orgfile = path
+
+    # file-like object (e.g. io.StringIO)
+    try:
+        # This will raise an AttributeError if it's not a file-like object.
+        all_lines = orgfile.readlines()
+        all_lines = (line.rstrip('\n') for line in all_lines)
```

Comment:
One letter variable names like `l` influencing the readability and maintainability of code negative. That is why I modified the variable names here to make it clearer and less error prone.